### DISCUSSION TITLE
/articles/sharepoint-framework-v1-20-whats-in-latest-update-of-spfx/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#314] [Comments]
Learn what I found while picking apart the latest SharePoint Framework (SPFx) release, v1.20. There’s not much in this release, but enough worth covering.
https://www.andrewconnell.com/articles/sharepoint-framework-v1-20-whats-in-latest-update-of-spfx/

**Comment by ynot3363** [comment]
I am a bit shocked that you didn't bring up another annoying fact that they need to fix with SPFx and that is the target is set to es5 and es6 is unsupported by them.

**Reply by andrewconnell** [reply]
Adding that here for tracking... great point @ynot3363! https://github.com/SharePoint/sp-dev-docs/issues/8262

**Reply by mikezimm** [reply]
Yes please!  I have been manually updating the target library to es6 for a couple years now and that one little number at the end makes life so much easier!

**Comment by kbeeveer46** [comment]
I look forward to your articles after every release! Thanks for doing them.

**Reply by andrewconnell** [reply]
Thanks!

**Comment by garth-mcmillan** [comment]
Regarding the SPHttpClientBatch API, previously you could refer to the relevant -plusbeta version of the module, such as "@microsoft/sp-core-library@1.12.1-plusbeta" to use it. Since 12.1 they don't seem to release them. 

I did read that you should now just reference the specific beta version but I've tried both "@microsoft/sp-core-library@1.14.0-rc.2" and "@microsoft/sp-core-library@1.14.0-beta.5" and neither appear to contain it.

For context I'm in the process of updating a SPFx package from 1.7.1. Currently working up to 1.14.0 (but will continue onwards).

My questions:

1. Do you have any suggestions as to what versions should be used going forward, as in what contains the beta exports?
1. If not, do you recommend we give up trying to use batching for the time being?

Your time is much appreciated. Thank you.

**Reply by andrewconnell** [reply]
@garth-mcmillan said:

> Regarding the SPHttpClientBatch API, previously you could refer to the relevant -plusbeta version of the module, such as "@microsoft/[sp-core-library@1.12.1-plusbeta](mailto:sp-core-library@1.12.1-plusbeta)" to use it. Since 12.1 they don't seem to release them.

Correct... they **finally** moved away from that unique way of rolling out preview features and instead, used the generally accepted practice of releasing `next` versions with NPM. To install beta versions, use the `@next` tag. For instance:

```bash
npm install @microsoft/generator-sharepoint@next --global
```

@garth-mcmillan said:

> Do you have any suggestions as to what versions should be used going forward, as in what contains the beta exports?

Things are listed as beta, or developer preview, in the production packages... they aren't relegated to specific packages.

@garth-mcmillan said:

> If not, do you recommend we give up trying to use batching for the time being?

For batching, there's nothing in the SPFx API that will help you. You can manually create the batch, *as I explain in the two articles I linked to above*, or use [Microsoft Graph's support for batch](https://learn.microsoft.com/graph/json-batching) ([SDK options](https://learn.microsoft.com/graph/sdks/batch-requests)) to call the SharePoint REST API.

**Reply by garth-mcmillan** [reply]
Thank you @andrewconnell for such a quick response and for such a good blog (I only just discovered it).

Unfortunately @microsoft/generator-sharepoint@next doesn't appear to install the preview features (at least not with 1.14.0).

I had noted your articles on batching and I will look at doing something similiar.

Thanks again.

**Reply by andrewconnell** [reply]
@garth-mcmillan said:
> Unfortunately @microsoft/generator-sharepoint@next doesn't appear to install the preview features (at least not with 1.14.0).

Right... when I said *Things are listed as beta, or developer preview, in the production packages... they aren't relegated to specific packages.*. 

In other words, there's no separate package, rather things are released in different levels of support (beta / developer preview) in the main package.
